building in the enterprise (#13690)
gatsbyjs/gatsby: docs/docs/building-in-the-enterprise.md

---
title: "Building With Gatsby in the Enterprise"
---

Some Gatsby advocates may be working inside “enterprise” software development organizations. Others may work at agencies which work closely with such organizations.

If this describes you, you have unique challenges, but also opportunities, building with Gatsby at your company.

This guide is designed to help you navigate the terrain in order to effectively advocate for Gatsby within your organization. 

### Common characteristics of enterprises

People mean different things when they say “enterprise”. Some mean 500-person companies. Other people mean 20,000-person companies. 

When we say “enterprise” we mean large companies with established infrastructural setups -- in-house layers of code, frameworks, processes, and preferred vendors to handle concerns such as hosting, deployments, CDN, etc.

> Other common characteristics include specified procurement processes, standardization on particular technology stacks, changeover at eg 5-year intervals, and frequent security audits. Often, enterprises are heavily invested in maintaining and improving existing web properties (“brownfield”) as opposed to creating new ones (“greenfield”).

Reviewer: > changeover at eg 5-year intervals

This is a little unclear to me. Is that supposed to be `e.g.`?

Author: Yep! Fixed.

### Key benefits of Gatsby for enterprises

One of the key challenges that enterprises face is keeping their tech stack up to date in order to enjoy productivity and performance benefits, while preventing their tooling setups from degrading into “legacy” code. 

A related challenge is how to continue to attract and retain top engineering talent.

> Surveys at enterprises adopting Gatsby has found that developers there highlight benefits such as:

Reviewer: ```suggestion
Surveys at enterprises adopting Gatsby have found that developers there highlight benefits such as:
```

**1. The ability to build with modern tools such as React, Webpack, and GraphQL.** Adopting more common frameworks, rather than custom in-house tooling, tends to increases developer satisfaction, developer productivity, and the ability to recruit developers. 

> **2. The ability to be productive from the start.** Reduced time on dev and build tolling setup / configuration helps developers get going right away. The burden of maintaining high-quality tooling setups is offloaded from internal tooling teams (and managers) to the Gatsby maintainer team.

Reviewer: >  Reduced time on dev and build tolling setup

build tooling setup?

Author: Oops :)

### Overcoming obstacles to adoption

Introducing Gatsby in the enterprise has unique challenges. After watching different sizes and types of enterprises adopt Gatsby, we’ve learned some lessons about how to overcome them. 

Depending on your type of your organization, not all of these challenges may apply to you!

**Challenge 1: Requirements to integrate with in-house tooling.** Some enterprises require integration with concerns such as those listed above. 

> A somewhat complete list of potential areas that may have in-house, standard solutions: auth, SSO, error logging, deployments, staging, hosting, security, CDN, middleware (data fetching), analytics, and internationalization/localization.

Reviewer: ```suggestion
A somewhat complete list of potential areas that may have in-house, standard solutions: authentication, SSO, error logging, deployments, staging, hosting, security, CDN, middleware (data fetching), analytics, and internationalization/localization.
```

**Challenge 2: Need for buy-in from executives.** Sometimes new technology adoption in the enterprise requires buy-in from executives who aren’t end-users of Gatsby. 

**Solution 2: Potential strategies to get buy-in include building consensus among developers, executing a simple proof of concept, and using the language of execs and managers.**
 
Depending on what your enterprise looks like, there can be different strategies for generating buy-in. These strategies can be used separately or together.

**Build consensus among developers.** If developers in your organization are influential, it can be helpful to build consensus among developers and engineering managers around Gatsby. If other developers aren’t familiar with Gatsby, holding “brown bag lunches” and workshops (going through the Tutorial tends to work well) can build excitement.

**Execute a simple proof of concept.** Actually having built something with Gatsby often changes the conversation — it becomes less of an abstract concept to others, and more of a concrete possible choice. Choosing a small project and building it with Gatsby can go a long way towards socializing Gatsby in your organization.

> **Use the language of executives and managers.** Executives have different goals and objectives than developers. Read the [Executive persona](/docs/winning-over-executives/) to see how to explain Gatsby in a way that executives find attractive. 

Reviewer: ```suggestion
**Use the language of executives and managers.** Executives have different goals and objectives than developers. Read about the [Executive persona](/docs/winning-over-executives/) to see how to explain Gatsby in a way that executives find attractive. 
```

One key strategy can be to align yourself to organization-wide “digital transformation” initiatives, especially if your company is in an industry that is being changed by the Internet (such as retail, travel, consumer goods, content, or hospitality). 

**Challenge 3: Required coordination with third-party agencies.** Some enterprises contract with third-party agencies for a moderate or significant part of their website development. Desires by an enterprise for technology shifts and upgrades often require tight coordination with agency partners. If an enterprise and its agency partners don’t have a strong relationship, coordinating on a switch to Gatsby becomes difficult. 

**Solution 3: Work closely with agency partners to ensure adopting Gatsby helps them, too.** It’s often a benefit to agencies to be familiar with cutting-edge technologies such as Gatsby (and the whole Gatsby stack: React, GraphQL, webpack, etc). It allows them to share their expertise with other client organizations. In addition, it helps them distinguish themselves from their competitors (other agencies). 

> Often, there are key individuals at agencies that are excited for personal and professional reasons to spearhead these kinds of projects. Identifying these individuals are working closely with them can unlock key doors and generate momentum to move your Gatsby project forward. 

Reviewer: ```suggestion
Often, there are key individuals at agencies that are excited for personal and professional reasons to spearhead these kinds of projects. Identifying these individuals and working closely with them can unlock key doors and generate momentum to move your Gatsby project forward. 
```

> One key to moving a Gatsby project is having a pre-existing relationship of trust between the enterprise and its agency partners. When this isn’t true — perhaps because of past behavior on the part of one or both parties, any sort of change becomes difficult. 

Reviewer: ```suggestion
One key to successfully completing a Gatsby project is having a pre-existing relationship of trust between the enterprise and its agency partners. When this isn’t true — perhaps because of past behavior on the part of one or both parties — any sort of change becomes difficult. 
```

### Conclusion

Adopting Gatsby in an enterprise can be more difficult than in small companies. However, there are lasting benefits for the company as well as the individuals guiding the change.

Because technology choices are more sticky in the enterprise, driving adoption of Gatsby may mean that you drive value for your company in a way that lasts for 5-10 years. 

The benefits your company will derive, such as performance and the ability to rapidly iterate on its consumer-facing web presence and digital footprint, will provide significant benefits in the marketplace. Your company will be more likely to thrive _because of the work you did_.

> Typically, individuals who guide and lead a shift to Gatsby are seen as forward-thinking and as agents of change, which can be [very beneficial for their careers within the company](https://www.gatsbyjs.org/docs/how-gatsby-boosts-career/). In addition, as the Gatsby rollout gains steam and proceeds through the company, they are prime candidates to spearhead the effort.

Reviewer: ```suggestion
Typically, individuals who guide and lead a shift to Gatsby are seen as forward-thinking and as agents of change, which can be [very beneficial for their careers within the company](/docs/how-gatsby-boosts-career/). In addition, as the Gatsby rollout gains steam and proceeds through the company, they are prime candidates to spearhead the effort.
```
